### PR TITLE
Add EIP: ERC-20Spendable

### PR DESCRIPTION
**Pull request by omnus**
Hi!

I've followed the template at https://github.com/ethereum/EIPs/blob/master/eip-template.md, hopefully I have included everything you need.

Thanks for taking the time to review it.

omnus

### CODE REVIEW

File: EIPS/eip-7777.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-08-17
+requires: 20
+---
+
+## Abstract
+
+An extension of ERC20 that allows user's to initiate a transfer of tokens that:
```
Comment (abcoathup) [inline EIPS/eip-7777.md:16]:
```suggestion
An extension of ERC-20 that allows user's to initiate a transfer of tokens that:
```

File: EIPS/eip-7777.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+1. Transfers the tokens.
+2. Calls a hook in the receiver passing arguments.
+3. Receives returned arguments.
+
+This allows contract interations that are currently performed with the approve -> pull pattern to be executed with a single EOA method call.
+
+## Motivation
+
+Any ERC20 interaction beyond a simple transfer requires the user to authorise a contract to their token then make a call on this contract. This has two main disadvantages:
```
Comment (abcoathup) [inline EIPS/eip-7777.md:26]:
```suggestion
Any ERC-20 interaction beyond a simple transfer requires the user to authorise a contract to their token then make a call on this contract. This has two main disadvantages:
```

File: EIPS/eip-7777.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+1. The holder of the ERC20 token must make two contract calls for a single operation. This increases friction and gas cost.
```
Comment (abcoathup) [inline EIPS/eip-7777.md:28]:
```suggestion
1. The holder of the ERC-20 token must make two contract calls for a single operation. This increases friction and gas cost.
```

File: EIPS/eip-7777.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+2. The holder gives permission to transfer tokens (in some cases for all of their holding), to another address. This has obvious security implications, and is the root cause of many stolen tokens.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Definitions
+
+- ERC20Spendable: A ERC20 implementing the ERC20Spendable standard. This ERC20 gains the _spend_ method, where the caller specifies the recipient, amount to transfer and any arguments encoded into bytes.
```
Comment (abcoathup) [inline EIPS/eip-7777.md:37]:
```suggestion
- ERC20Spendable: A ERC-20 implementing the ERC20Spendable standard. This ERC-20 gains the _spend_ method, where the caller specifies the recipient, amount to transfer and any arguments encoded into bytes.
```

File: EIPS/eip-7777.md [hunk 5]
```diff
@@ -0,0 +1,341 @@
+---
+eip: 7777
```
Comment (abcoathup) [inline EIPS/eip-7777.md:2]:
```suggestion
eip: 7492
```
Assigning EIP number as PR number.  Please also rename the filename.

File: EIPS/eip-7777.md (hunk 1 above)
Comment (omnus) [reply EIPS/eip-7777.md:16]: Ooh, great catch, thank you! I forgot the hyphen was back in....

File: EIPS/eip-7777.md (hunk 5 above)
Comment (omnus) [reply EIPS/eip-7777.md:2]: Ah, so that's how I work out the eip number, thank you!

**Comment by abcoathup** [comment]
Please update the filename eip-7777.md to the EIP number eip-7492.md

**Review by jonesk7734 — APPROVED** [review]
😀

File: EIPS/eip-7492.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
+---
+eip: 7492
+title: ERC-20Spendable
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:3]: This title is not descriptive. Please use a title like "ERC-20 <whatever> Extension".

File: EIPS/eip-7492.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Definitions
+
+- `ERC20Spendable`: A [ERC-20](./eip-20.md) implementing the `ERC20Spendable` standard. This [ERC-20](./eip-20.md) gains the _spend_ method, where the caller specifies the recipient, amount to transfer and any arguments encoded into bytes.
+- `ERC20SpendableReceiver`: A contract implementing `ERC20SpendableReceiver`. This contract gains the `onERC20SpendableReceived` method to receive spend requests.
+- spend: The new method on the [ERC-20](./eip-20.md) to transfer tokens, call the receiver, and handle returned arguments.
+- `onERC20SpendableReceived`: The new method on the receiver to handle _spend_ requests and return arguments.
+
+### Solidity API
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:42]:
```suggestion
### Solidity ABI
```
ABI is the correct term to use here

File: EIPS/eip-7492.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+1. The holder of the [ERC-20](./eip-20.md) token must make two contract calls for a single operation. This increases friction and gas cost.
+2. The holder gives permission to transfer tokens (in some cases for all of their holding), to another address. This has obvious security implications, and is the root cause of many stolen tokens.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Definitions
+
+- `ERC20Spendable`: A [ERC-20](./eip-20.md) implementing the `ERC20Spendable` standard. This [ERC-20](./eip-20.md) gains the _spend_ method, where the caller specifies the recipient, amount to transfer and any arguments encoded into bytes.
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:37]: This definition is self-referencing.

File: EIPS/eip-7492.md [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+- `ERC20SpendableReceiver`: A contract implementing `ERC20SpendableReceiver`. This contract gains the `onERC20SpendableReceived` method to receive spend requests.
+- spend: The new method on the [ERC-20](./eip-20.md) to transfer tokens, call the receiver, and handle returned arguments.
+- `onERC20SpendableReceived`: The new method on the receiver to handle _spend_ requests and return arguments.
+
+### Solidity API
+
+#### `ERC20Spendable`
+
+Implementation of `ERC20Spendable`.
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:46]:
```suggestion
```
This looks like placeholder text you forgot to remove.

File: EIPS/eip-7492.md [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+
+`ERC20Spendable` allows [ERC-20](./eip-20.md)s to operate as 'spendable' items, i.e. an [ERC-20](./eip-20.md) token that
+can trigger an action on another contract at the same time as being transfered. Similar to `ERC-677`
+and the hooks in [ERC-777](./eip-777.md), but with more of an empasis on interoperability (returned values) than
+`ERC-677` and specifically scoped interaction rather than the general hooks of [ERC-777](./eip-777.md).
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:51]: ERC-677 does not exist.

This is also Rationale, not Specification.

File: EIPS/eip-7492.md [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+This method will pass back the valid bytes4 selector and any bytes argument passed from
+{\_handleSpend}.
+
+###### Parameters
+
+| Name        | Type    | Description                               |
+| ----------- | ------- | ----------------------------------------- |
+| spender\_   | address | The address spending the `ERC20Spendable` |
+| spent\_     | uint256 | The amount of token spent                 |
+| arguments\_ | bytes   | Bytes sent with the call                  |
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:309]: Why don't you just use a solidity interface (like your IERC20Spendable.sol!) annotated with NatSpec as your specification? You're overcomplicating things here. See [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155#specification) for an example of a more modern spec.

File: EIPS/eip-7492.md [hunk 12]
```diff
@@ -0,0 +1,333 @@
...
+
+The full SDLC for this proposal has been completed. The contract source code and unit test suite is available in `../assets/eip-7492/`, as is the source code and example (mock) implementations.
+
+## Reference Implementation
+
+Please see `../assets/eip-7492/contracts`
+
+## Security Considerations
+
+Potential negative security implications have been considered and none are envisaged. This EIP is designed to improve end-user security by the reduction in number of contract approvals required.
```
Comment (Pandapip1) [inline EIPS/eip-7492.md:329]:
```suggestion
Needs discussion.
```
This is a draft EIP. Unless you've got this thing audited, discussion should be solicited.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.

**Pull request closed without merging**